### PR TITLE
`ApplySchema`: support `--batch-size` flag in 'direct' strategy

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

This PR offers a new `--batch-size` flag to `ApplySchema`, like so:

```shell
vtctldclient ApplySchema --sql-file /tmp/tables.sql --batch-size=200 commerce
```

The flag `--batch-size` is only allowed when:

- DDL strategy is `direct` (which is also the default if undefined)
- `--uuid-list` is empty
- All SQL statements are `CREATE TABLE|VIEW`. No `ALTER`, no `DROP`, etc.

Under those conditions (and `ApplySchema` fails if any are unmet), the queries in `--sql` or `--sql-file` are sent to the backend shards in batches. In the above example, queries are sent in batches of `200` statements at a time.

Batching queries reduces network overhead. We have observed the overhead to be dramatic when the number of SQL statements is high. In a local test environment, where network latency is low, and in a 4000 tables schema, we saw a decrease from `3min` without batching to `1sec` with `--batch-size-500`. The actual optimal values may depend on the actual table definitions (as in number of bytes sent) and network latency. This is why `vitess` does not pick an arbitrary value and lets the user choose the batch size.

The use case is for when one wishes to apply a large number (e.g. thousands) of changes. This has one common scenario, which is populating a brand new, empty shard, with a known schema. To that effect we currently restrict `--batch-size` to only support `CREATE` statements. Strictly speaking batching could also work for `ALTER|DROP` statements, but we first want to address this particular use case and based on our learnings we may remove the query restriction.

Unit tests and endtoend tests added.

Docs TBD

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/13692

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Review by dbussink — APPROVED** [review]
This is a great performance improvement!

File: go/cmd/vtctldclient/command/schema.go [hunk 1]
```diff
@@ -294,6 +296,7 @@ func init() {
 	ApplySchema.Flags().StringVar(&applySchemaOptions.CallerID, "caller-id", "", "Effective caller ID used for the operation and should map to an ACL name which grants this identity the necessary permissions to perform the operation (this is only necessary when strict table ACLs are used).")
 	ApplySchema.Flags().StringArrayVar(&applySchemaOptions.SQL, "sql", nil, "Semicolon-delimited, repeatable SQL commands to apply. Exactly one of --sql|--sql-file is required.")
 	ApplySchema.Flags().StringVar(&applySchemaOptions.SQLFile, "sql-file", "", "Path to a file containing semicolon-delimited SQL commands to apply. Exactly one of --sql|--sql-file is required.")
+	ApplySchema.Flags().Int64Var(&applySchemaOptions.BatchSize, "batch-size", 0, "How many queries to batch together")
```
[the remark this answers is not in the document]
Comment (shlomi-noach) [reply go/cmd/vtctldclient/command/schema.go:299]: Done

**Review by rohit-nayak-ps — APPROVED** [review]

**Comment by shlomi-noach** [comment]
Merging, then creating a docs PR.

**Comment by shlomi-noach** [comment]
Docs: https://github.com/vitessio/website/pull/1556

**Pull request merged**